x-pack/legacy/plugins/snapshot_restore/public/app/components/restore_snapshot_form/steps/step_logistics.tsx

```diff
@@ -55,11 +56,25 @@ export const RestoreSnapshotStepLogistics: React.FunctionComponent<StepProps> =
       (index): Option => ({
         label: index,
         checked:
-          isAllIndices || (restoreIndices && restoreIndices.includes(index)) ? 'on' : undefined,
+          isAllIndices ||
+          typeof restoreIndices === 'string' ||
+          (Array.isArray(restoreIndices) && restoreIndices.includes(index))
+            ? 'on'
+            : undefined,
       })
     )
   );
 
+  // State for using selectable indices list or custom patterns
+  const [selectIndicesMode, setSelectIndicesMode] = useState<'list' | 'custom'>(
+    typeof restoreIndices === 'string' || snapshotIndices.length > 100 ? 'custom' : 'list'
```

Comment:
I love the UX consideration we're making here. I think it warrants a comment, e.g.

```js
// Users with more than 100 indices will probably want to use an index pattern to select
// them instead, so we'll default to showing them the index pattern input.
```

Also, the Index Pattern Wizard shows you a preview of matching indices to your index pattern. Do you think users would benefit from something similar here? Even if so, I would not consider it a blocker for this PR.

Reply:
BTW if we do decide to fetch matching indices at some point, https://github.com/elastic/kibana/issues/42045 contains a suggestion on how to make that request more efficient.